### PR TITLE
Add basics for Vcards

### PR DESCRIPTION
**Pull request by DimitrijeDimitrijevic**
Added parsing xml response from server for contacts with addressbook and contact structs. 
Added also basic CRUD operations on contacts.

### CODE REVIEW

File: lib/excalt/vcard/contact.ex [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+       {:ok, %Finch.Response{status: status, body: body}} ->
+         IO.inspect(status: status)
+         IO.inspect(body: body)
+     end
+    end
+
+  def delete(server_url, username, password, addressbook_name, contact_uuid) do
+    authentication = Excalt.Vcard.Addressbook.build_authentication_header(username, password)
+    req_url = Excalt.Vcard.UrlHelper.build_url(server_url, username, addressbook_name)
+    req_url = "#{req_url}" <> "/" <> "#{contact_uuid}.vcf"
```
Comment (swerter) [inline lib/excalt/vcard/contact.ex]: I would simply use:

    "#{req_url}/#{contact_uuid}.vcf"

File: lib/excalt/vcard/contact.ex [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+      _ ->
+        :error
+    end
+   end
+ # Client need to change the whole vcard, no way of incremental changes: https://www.rfc-editor.org/rfc/rfc6352#section-9.1
+  def update(server_url, username, password, addressbook_name, etag, contact_uuid, vcf_text) do
+    authentication = Excalt.Vcard.Addressbook.build_authentication_header(username, password)
+    req_url = Excalt.Vcard.UrlHelper.build_url(server_url, username, addressbook_name)
+    req_url = "#{req_url}" <> "/" <> "#{contact_uuid}.vcf"
+    IO.inspect(req_url: req_url)
```
Comment (swerter) [inline lib/excalt/vcard/contact.ex]: please remove all IO.inspects and co.

And more generally, please run `mix format` and `mix credo` before making the pull request. Thanks.

File: lib/excalt/xml/builder.ex [hunk 3]
```diff
@@ -20,10 +22,15 @@ defmodule Excalt.XML.Builder do
...
+      |> Saxy.encode!([])
 
-    Saxy.encode!(el, [])
   end
 
+  @doc """
+  Builds XML request for listing events from a calendar collections.
+  Gets only the VEVENT in VCALENDAR object.
+  """
+  @spec event_list() :: String.t()
```
Comment (swerter) [inline lib/excalt/xml/builder.ex:32]: Nice, thanks for adding the specs and docs!

File: test/excalt/builder_test.exs [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    assert request == Builder.calendar_list
+  end
+
+  describe "XML request for events" do
+    test "all events" do
+
+    end
+    test "timerange events"
+  end
+end
```
Comment (swerter) [inline test/excalt/builder_test.exs:33]: I think there could be some additional tests ;)

**Review by swerter — COMMENTED** [review]
Thanks for your work. Looks good from the top of my head. Just a few small comments.

File: test/excalt/builder_test.exs (hunk 4 above)
Comment (DimitrijeDimitrijevic) [reply test/excalt/builder_test.exs:33]: I will do more testing, on calendar and on vcards.

**Pull request merged**